numpy/random/src/distributions/distributions.c

```diff
@@ -943,7 +943,7 @@ RAND_INT_TYPE random_geometric_search(bitgen_t *bitgen_state, double p) {
 }
 
 RAND_INT_TYPE random_geometric_inversion(bitgen_t *bitgen_state, double p) {
-  return (RAND_INT_TYPE)ceil(log(1.0 - next_double(bitgen_state)) / log(1.0 - p));
+  return (RAND_INT_TYPE)ceil(npy_log1p(-next_double(bitgen_state)) / npy_log1p(-p));
```

Comment:
This can still overflow if `p < float.fromhex('-0x1.25e4f7b2737fbp-1019')`, but that's a lot smaller than the `10**-20` in the original issue.

Reply:
This one also could break the stream I think due to the second argument, and so would need a new function.